x-pack/solutions/security/plugins/security_solution/public/flyout/document_details/right/components/highlighted_fields_cell.tsx

```diff
@@ -62,14 +68,84 @@ export const HighlightedFieldsCell: FC<HighlightedFieldsCellProps> = ({
   scopeId = '',
   showPreview = false,
   ancestorsIndexName,
+  displayValuesLimit = 2,
 }) => {
   const agentType: ResponseActionAgentType = useMemo(() => {
     return getAgentTypeForAgentIdField(originalField);
   }, [originalField]);
   const { euiTheme } = useEuiTheme();
+  const [isContentExpanded, setIsContentExpanded] = useState(false);
+  const toggleContentExpansion = useCallback(
+    () => setIsContentExpanded((currentIsOpen) => !currentIsOpen),
+    []
+  );
+
+  const visibleValues = useMemo(() => {
+    /**
+     * Check if a limit was set and if the limit
+     * is within the values size
+     */
+    if (
+      displayValuesLimit &&
+      displayValuesLimit > 0 &&
+      displayValuesLimit < (values?.length ?? 0)
+    ) {
+      return values?.slice(0, displayValuesLimit);
+    }
+
+    return values;
+  }, [values, displayValuesLimit]);
+
+  const overflownValues = useMemo(() => {
+    /**
+     * Check if a limit was set and if the limit
+     * is within the values size
+     */
+    if (
+      displayValuesLimit &&
+      displayValuesLimit > 0 &&
+      displayValuesLimit < (values?.length ?? 0)
+    ) {
+      return values?.slice(displayValuesLimit);
+    }
+
+    return [];
```

Comment:
Super nit, we should create a `const emptyArray = [];` above the component in this file and use it here. That way, if for some reason `values` or `displayValuesLimit` change but the `if` condition above is still not true, we won't trigger any rerenders